test/database.js

```diff
@@ -59,6 +59,19 @@ describe('database', function() {
     await dbShort.close();
   });
 
+  it('should accept connection with options', async () => {
+    // auth should fail with invalid 'authSource', to make sure options take effect
+    const cannotConnect = mongoist('localhost:27017/test', { 'authSource': 'dummy' });
+    try {
+      await cannotConnect.test.find();
+      await cannotConnect.close();
+      expect(false, "the error should occurred before");
+    } catch (e) {
+      expect(e).to.exist;
+      return;
+    }
```

Comment:
Could we refine this a bit? Maybe something like:
```suggestion
    const cannotConnect = mongoist('localhost:27017/test', { 'authSource': 'invalid_source' });
    let err;
    try {
      await cannotConnect.test.find();
      await cannotConnect.close();
    } catch (e) {
      err = e;
    }
    
    expect(e).to.exist;
```

Reply:
Thanks for the reviewing, a new commit has been submitted which refines the test case according to your suggestion.